rflx/expression.py

```diff
@@ -474,6 +524,17 @@ def neutral_element(self) -> int:
     def symbol(self) -> str:
         return ' + '
 
+    def z3expr(self) -> z3.ArithRef:
+        z3expr = self.terms[0].z3expr()
+        for t in self.terms[1:]:
+            tmp = t.z3expr()
+            if not isinstance(z3expr, z3.ArithRef) or not isinstance(tmp, z3.ArithRef):
+                raise ValueError
+            z3expr = z3expr + tmp
```

Comment:
What is the difference to just using `z3expr = sum(t.z3expr() for t in self.terms)` instead? (same below)

Reply:
Using sum() is a good idea (cf. e332023). Unfortunately there is [no equivalent](https://stackoverflow.com/questions/595374/whats-the-function-like-sum-but-for-multiplication-product) for multiplication for Python < 3.8